webapp/packages/plugin-data-viewer-conditional-formatting/src/getMiddleColor.ts

```diff
@@ -0,0 +1,42 @@
+/*
+ * CloudBeaver - Cloud Database Manager
+ * Copyright (C) 2020-2025 DBeaver Corp and others
+ *
+ * Licensed under the Apache License, Version 2.0.
+ * you may not use this file except in compliance with the License.
+ */
+
+export function getMiddleColor(minColor: string, maxColor: string): string {
+  // Create a temporary canvas to parse CSS colors
+  const canvas = document.createElement('canvas');
+  canvas.width = canvas.height = 1;
+  const ctx = canvas.getContext('2d', { willReadFrequently: true });
+
+  if (!ctx) {
+    return minColor;
+  }
+
+  // Parse first color
+  ctx.fillStyle = minColor;
+  ctx.fillRect(0, 0, 1, 1);
+  const data1 = ctx.getImageData(0, 0, 1, 1).data;
+  const r1 = data1[0] ?? 0;
+  const g1 = data1[1] ?? 0;
+  const b1 = data1[2] ?? 0;
+
+  // Parse second color
+  ctx.fillStyle = maxColor;
+  ctx.fillRect(0, 0, 1, 1);
+  const data2 = ctx.getImageData(0, 0, 1, 1).data;
+  const r2 = data2[0] ?? 0;
+  const g2 = data2[1] ?? 0;
+  const b2 = data2[2] ?? 0;
```

Comment:
Do we really need those 0 fallbacks?

Looks like repetition and we could wrap it in parse function with color argument. Plus use `[r ,g ,b] = ctx....data`